lib/txnManager.js

```diff
@@ -0,0 +1,148 @@
+const assert = require('bsert');
+import TXUX from './uxtx.js';
+
+// importing to bundle with TxnManagerOptions
+import UXTXOptions from './uxtxOptions';
+
+class TxnManager {
+  constructor(options) {
+    // hash -> parsed transaction
+    this._parsed = {};
+
+    this._constants = null;
+    this._custom = null;
+    this._labels = null;
+
+    if (options)
+      this.fromOptions(options);
+  }
+
+  static fromOptions(options) {
+    return new this().fromOptions(options);
+  }
+
+  /**
+   * Inject properties from options object.
+   * @private
+   * @param {Object} options
+   * @param {Object} options.constants - internal constants
+   * @param {Object} options.labels
+   *   human readable labels, used in toJSON output
+   */
+  fromOptions(options) {
+    assert(typeof options === 'object', 'must pass options object');
+    if (options.custom) {
+      assert(typeof options.custom === 'function');
+      this._custom = options.custom;
+    }
+
+    if (options.constants) {
+      // TODO: validation
+      this._constants = options.constants;
+    }
+
+    if (options.labels) {
+      // TODO: validation
+      this._labels = options.labels;
+    }
+
+    return this;
+  }
+
+  getConstants() {
+    assert(this._constants);
+    return this._constants;
```

Comment:
do the `get`s need to do an assert? wouldn't returning undefined (or it could be a ternary and return false) be fine?

Reply:
They are required for the proper functionality of the component which is why I am asserting for them here

Reply:
would they make more sense then in the instantiation or `fromOptions` then?

Reply:
👍